File: targets/java-spring/controllers/ToolController.java
package com.baddemo.controllers;

import javax.script.ScriptEngine;
import javax.script.ScriptEngineManager;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.baddemo.services.ToolService;

@RestController
@RequestMapping("/tools")
public class ToolController {

    private final ToolService toolService = new ToolService();

    @GetMapping("/ping")
    public String ping(@RequestParam String host) throws Exception {
        Process p = Runtime.getRuntime().exec("ping -c 1 " + host);
        return "rc=" + p.waitFor();
    }

    @GetMapping("/diagnose")
    public String diagnose(@RequestParam String host) throws Exception {
        toolService.stageTarget(host);
        return "rc=" + toolService.runStagedDiag();
    }

    @PostMapping("/calc")
    public String calc(@RequestBody String expr) throws Exception {
        ScriptEngine engine = new ScriptEngineManager().getEngineByName("JavaScript");
        Object result = engine.eval(expr);
        return String.valueOf(result);
    }
}

File: targets/php-laravel/app/Http/Controllers/ToolController.php
<?php

namespace App\Http\Controllers;

use App\Services\ToolService;
use Illuminate\Http\Request;

class ToolController
{
    private ToolService $tools;

    public function __construct()
    {
        $this->tools = new ToolService();
    }

    public function ping(Request $request)
    {
        $host = $request->query('host', '');
        $rc = system("ping -c 1 " . $host);
        return response()->json(['rc' => $rc]);
    }

    public function diagnose(Request $request)
    {
        $host = $request->query('host', '');
        $this->tools->stageTarget($host);
        $output = $this->tools->runStagedDiag();
        return response()->json(['output' => $output]);
    }

    public function calc(Request $request)
    {
        $expr = $request->input('expr', '0');
        $result = eval("return " . $expr . ";");
        return response()->json(['result' => $result]);
    }
}
